authenticationManager must be specified

**Describe the bug**
The app run failed, with ERROR: Error creating bean with name 'prometheusAuthFilter' defined in file [**\nacos\prometheus\target\classes\com\alibaba\nacos\prometheus\filter\PrometheusAuthFilter.class]: authenticationManager must be specified

**Actually behavior**
env: 
    -Dnacos.standalone=true
debug main:
    NacosBootstrap

**Desktop (please complete the following information):**
 - OS: [windows11]
 - Version [3.0.0-alpha.2]
 - Module [bootstrap]
 - SDK [zulu-17]

**Additional context**
Detail:

2025-01-23T20:23:08.722+08:00  INFO 43280 --- [acos-starting.0] c.a.n.c.l.StartingApplicationListener    : Nacos Server is starting...
2025-01-23T20:23:09.733+08:00  INFO 43280 --- [acos-starting.0] c.a.n.c.l.StartingApplicationListener    : Nacos Server is starting...
2025-01-23T20:23:09.995+08:00  WARN 43280 --- [           main] s.c.a.AnnotationConfigApplicationContext : Exception encountered during context initialization - cancelling refresh attempt: org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'prometheusAuthFilter' defined in file [...\nacos\prometheus\target\classes\com\alibaba\nacos\prometheus\filter\PrometheusAuthFilter.class]: authenticationManager must be specified
2025-01-23T20:23:11.200+08:00  WARN 43280 --- [           main] c.a.nacos.sys.file.WatchFileCenter       : [WatchFileCenter] start close
2025-01-23T20:23:11.201+08:00  WARN 43280 --- [           main] c.a.nacos.sys.file.WatchFileCenter       : [WatchFileCenter] start to shutdown this watcher which is watch : \opt\nacos\conf
2025-01-23T20:23:11.201+08:00  WARN 43280 --- [           main] c.a.nacos.sys.file.WatchFileCenter       : [WatchFileCenter] already closed
2025-01-23T20:23:11.201+08:00  INFO 43280 --- [           main] c.a.nacos.common.notify.NotifyCenter     : [NotifyCenter] Start destroying Publisher
2025-01-23T20:23:11.201+08:00  INFO 43280 --- [           main] c.a.nacos.common.notify.NotifyCenter     : [NotifyCenter] Completed destruction of Publisher
2025-01-23T20:23:11.201+08:00 ERROR 43280 --- [           main] c.a.n.c.l.StartingApplicationListener    : Startup errors : 

org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'prometheusAuthFilter' defined in file [...\nacos\prometheus\target\classes\com\alibaba\nacos\prometheus\filter\PrometheusAuthFilter.class]: authenticationManager must be specified
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.initializeBean(AbstractAutowireCapableBeanFactory.java:1808)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:601)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:523)
	at org.springframework.beans.factory.support.AbstractBeanFactory.lambda$doGetBean$0(AbstractBeanFactory.java:336)
	at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:289)
	at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:334)
	at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:199)
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.instantiateSingleton(DefaultListableBeanFactory.java:1122)
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.preInstantiateSingleton(DefaultListableBeanFactory.java:1093)
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.preInstantiateSingletons(DefaultListableBeanFactory.java:1030)
	at org.springframework.context.support.AbstractApplicationContext.finishBeanFactoryInitialization(AbstractApplicationContext.java:987)
	at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:627)
	at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:752)
	at org.springframework.boot.SpringApplication.refreshContext(SpringApplication.java:439)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:318)
	at org.springframework.boot.builder.SpringApplicationBuilder.run(SpringApplicationBuilder.java:149)
	at com.alibaba.nacos.bootstrap.NacosBootstrap.startCoreContext(NacosBootstrap.java:92)
	at com.alibaba.nacos.bootstrap.NacosBootstrap.main(NacosBootstrap.java:46)
Caused by: java.lang.IllegalArgumentException: authenticationManager must be specified
	at org.springframework.util.Assert.notNull(Assert.java:181)
	at org.springframework.security.web.authentication.AbstractAuthenticationProcessingFilter.afterPropertiesSet(AbstractAuthenticationProcessingFilter.java:191)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.invokeInitMethods(AbstractAutowireCapableBeanFactory.java:1855)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.initializeBean(AbstractAutowireCapableBeanFactory.java:1804)
	... 17 common frames omitted

2025-01-23T20:23:11.202+08:00 ERROR 43280 --- [           main] c.a.n.c.l.StartingApplicationListener    : Nacos failed to start, please see \opt\nacos\logs\nacos.log for more details.
2025-01-23T20:23:11.207+08:00  INFO 43280 --- [           main] .s.b.a.l.ConditionEvaluationReportLogger : 

Error starting ApplicationContext. To display the condition evaluation report re-run your application with 'debug' enabled.
2025-01-23T20:23:11.220+08:00 ERROR 43280 --- [           main] o.s.boot.SpringApplication               : Application run failed

org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'prometheusAuthFilter' defined in file [...\nacos\prometheus\target\classes\com\alibaba\nacos\prometheus\filter\PrometheusAuthFilter.class]: authenticationManager must be specified
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.initializeBean(AbstractAutowireCapableBeanFactory.java:1808)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:601)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:523)
	at org.springframework.beans.factory.support.AbstractBeanFactory.lambda$doGetBean$0(AbstractBeanFactory.java:336)
	at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:289)
	at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:334)
	at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:199)
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.instantiateSingleton(DefaultListableBeanFactory.java:1122)
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.preInstantiateSingleton(DefaultListableBeanFactory.java:1093)
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.preInstantiateSingletons(DefaultListableBeanFactory.java:1030)
	at org.springframework.context.support.AbstractApplicationContext.finishBeanFactoryInitialization(AbstractApplicationContext.java:987)
	at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:627)
	at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:752)
	at org.springframework.boot.SpringApplication.refreshContext(SpringApplication.java:439)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:318)
	at org.springframework.boot.builder.SpringApplicationBuilder.run(SpringApplicationBuilder.java:149)
	at com.alibaba.nacos.bootstrap.NacosBootstrap.startCoreContext(NacosBootstrap.java:92)
	at com.alibaba.nacos.bootstrap.NacosBootstrap.main(NacosBootstrap.java:46)
Caused by: java.lang.IllegalArgumentException: authenticationManager must be specified
	at org.springframework.util.Assert.notNull(Assert.java:181)
	at org.springframework.security.web.authentication.AbstractAuthenticationProcessingFilter.afterPropertiesSet(AbstractAuthenticationProcessingFilter.java:191)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.invokeInitMethods(AbstractAutowireCapableBeanFactory.java:1855)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.initializeBean(AbstractAutowireCapableBeanFactory.java:1804)
	... 17 common frames omitted

2025-01-23T20:23:11.221+08:00  INFO 43280 --- [       Thread-3] c.a.n.common.executor.ThreadPoolManager  : [ThreadPoolManager] Start destroying ThreadPool
2025-01-23T20:23:11.221+08:00  INFO 43280 --- [       Thread-3] c.a.n.common.executor.ThreadPoolManager  : [ThreadPoolManager] Completed destruction of ThreadPool

Process finished with exit code 1

--- a/console/src/main/java/com/alibaba/nacos/console/config/ConsoleWebConfig.java
+++ b/console/src/main/java/com/alibaba/nacos/console/config/ConsoleWebConfig.java
@@ -26,6 +26,7 @@ import org.springframework.boot.autoconfigure.jackson.Jackson2ObjectMapperBuilde
 import org.springframework.boot.web.servlet.FilterRegistrationBean;
 import org.springframework.context.annotation.Bean;
 import org.springframework.security.config.annotation.web.builders.HttpSecurity;
+import org.springframework.security.config.annotation.web.configurers.AbstractHttpConfigurer;
 import org.springframework.security.web.SecurityFilterChain;
 import org.springframework.stereotype.Component;
 import org.springframework.web.cors.CorsConfiguration;
@@ -137,7 +138,9 @@ public class ConsoleWebConfig {
     
     @Bean
     public SecurityFilterChain securityFilterChain(HttpSecurity http) throws Exception {
-        return http.authorizeHttpRequests().requestMatchers("/**").permitAll().and().csrf().disable().build();
+        http.authorizeHttpRequests((authorizeHttpRequests) -> authorizeHttpRequests.requestMatchers("/**").permitAll());
+        http.csrf(AbstractHttpConfigurer::disable);
+        return http.build();
     }
     
     public boolean isConsoleUiEnabled() {

--- a/console/src/main/java/com/alibaba/nacos/console/handler/config/ConfigHandler.java
+++ b/console/src/main/java/com/alibaba/nacos/console/handler/config/ConfigHandler.java
@@ -214,6 +214,7 @@ public interface ConfigHandler {
      * @param namespaceId  the namespaceId
      * @param remoteIp     the IP address of the client making the request
      * @param requestIpApp the name of the application making the request
+     * @param srcUser      the src user performing the operation
      * @return true if the beta configuration is successfully removed
      */
     boolean removeBetaConfig(String dataId, String group, String namespaceId, String remoteIp, String requestIpApp,

--- a/plugin-default-impl/nacos-default-auth-plugin/src/main/java/com/alibaba/nacos/plugin/auth/impl/configuration/NacosAuthPluginWebConfig.java
+++ b/plugin-default-impl/nacos-default-auth-plugin/src/main/java/com/alibaba/nacos/plugin/auth/impl/configuration/NacosAuthPluginWebConfig.java
@@ -29,6 +29,7 @@ import org.springframework.security.config.BeanIds;
 import org.springframework.security.config.annotation.authentication.configuration.AuthenticationConfiguration;
 import org.springframework.security.config.annotation.web.builders.HttpSecurity;
 import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
+import org.springframework.security.config.annotation.web.configurers.AbstractHttpConfigurer;
 import org.springframework.security.web.SecurityFilterChain;
 
 /**
@@ -62,8 +63,12 @@ public class NacosAuthPluginWebConfig {
         if (StringUtils.isBlank(ignoreUrls)) {
             return http.build();
         }
-        return http.authorizeHttpRequests().requestMatchers(ignoreUrls.trim().split(SECURITY_IGNORE_URLS_SPILT_CHAR))
-                .permitAll().and().csrf().disable().build();
+        final String finalIgnoreUrls = ignoreUrls;
+        http.authorizeHttpRequests((authorizeHttpRequests) ->
+                authorizeHttpRequests.requestMatchers(finalIgnoreUrls.trim().split(SECURITY_IGNORE_URLS_SPILT_CHAR)).permitAll()
+        );
+        http.csrf(AbstractHttpConfigurer::disable);
+        return http.build();
     }
     
     /**

--- a/prometheus/src/main/java/com/alibaba/nacos/prometheus/conf/PrometheusSecurityConfiguration.java
+++ b/prometheus/src/main/java/com/alibaba/nacos/prometheus/conf/PrometheusSecurityConfiguration.java
@@ -19,6 +19,7 @@ package com.alibaba.nacos.prometheus.conf;
 import com.alibaba.nacos.common.utils.StringUtils;
 import com.alibaba.nacos.core.web.NacosWebBean;
 import com.alibaba.nacos.plugin.auth.constant.Constants;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
 import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.Condition;
 import org.springframework.context.annotation.ConditionContext;
@@ -39,14 +40,16 @@ import static com.alibaba.nacos.prometheus.api.ApiConstants.PROMETHEUS_CONTROLLE
  */
 @Configuration
 @NacosWebBean
+@ConditionalOnProperty(name = "nacos.prometheus.metrics.enabled", havingValue = "true")
 public class PrometheusSecurityConfiguration {
     
     @Bean
     @Conditional(ConditionOnNoAuthPluginType.class)
     public SecurityFilterChain prometheusSecurityFilterChain(HttpSecurity http) throws Exception {
-        return http.authorizeHttpRequests()
-                .requestMatchers(PROMETHEUS_CONTROLLER_PATH, PROMETHEUS_CONTROLLER_NAMESPACE_PATH,
-                        PROMETHEUS_CONTROLLER_SERVICE_PATH).permitAll().and().getOrBuild();
+        http.authorizeHttpRequests(
+                (authorizeHttpRequests) -> authorizeHttpRequests.requestMatchers(PROMETHEUS_CONTROLLER_PATH,
+                        PROMETHEUS_CONTROLLER_NAMESPACE_PATH, PROMETHEUS_CONTROLLER_SERVICE_PATH).permitAll());
+        return http.getOrBuild();
     }
     
     private static class ConditionOnNoAuthPluginType implements Condition {

--- a/prometheus/src/main/java/com/alibaba/nacos/prometheus/filter/PrometheusAuthFilter.java
+++ b/prometheus/src/main/java/com/alibaba/nacos/prometheus/filter/PrometheusAuthFilter.java
@@ -16,18 +16,24 @@
 
 package com.alibaba.nacos.prometheus.filter;
 
+import com.alibaba.nacos.core.web.NacosWebBean;
 import com.alibaba.nacos.plugin.auth.constant.Constants;
+import com.alibaba.nacos.prometheus.controller.PrometheusController;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnBean;
 import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
 import org.springframework.boot.web.servlet.FilterRegistrationBean;
 import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.Configuration;
 import org.springframework.security.authentication.AuthenticationManager;
 import org.springframework.security.authorization.AuthenticatedAuthorizationManager;
+import org.springframework.security.config.annotation.authentication.builders.AuthenticationManagerBuilder;
+import org.springframework.security.config.annotation.web.builders.HttpSecurity;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.crypto.password.PasswordEncoder;
 import org.springframework.security.web.access.ExceptionTranslationFilter;
 import org.springframework.security.web.access.intercept.AuthorizationFilter;
 import org.springframework.security.web.authentication.AnonymousAuthenticationFilter;
 import org.springframework.security.web.authentication.Http403ForbiddenEntryPoint;
-import org.springframework.security.web.authentication.UsernamePasswordAuthenticationFilter;
 import org.springframework.security.web.authentication.www.BasicAuthenticationFilter;
 
 import static com.alibaba.nacos.prometheus.api.ApiConstants.PROMETHEUS_CONTROLLER_PATH;
@@ -37,9 +43,20 @@ import static com.alibaba.nacos.prometheus.api.ApiConstants.PROMETHEUS_CONTROLLE
  *
  * @author vividfish
  */
+@NacosWebBean
 @Configuration
 @ConditionalOnProperty(value = Constants.Auth.NACOS_CORE_AUTH_ENABLED, havingValue = "true")
-public class PrometheusAuthFilter extends UsernamePasswordAuthenticationFilter {
+@ConditionalOnBean(PrometheusController.class)
+public class PrometheusAuthFilter {
+    
+    @Bean
+    public AuthenticationManager authenticationManager(HttpSecurity http, UserDetailsService userDetailsService,
+            PasswordEncoder passwordEncoder) throws Exception {
+        AuthenticationManagerBuilder authenticationManagerBuilder = http.getSharedObject(
+                AuthenticationManagerBuilder.class);
+        authenticationManagerBuilder.userDetailsService(userDetailsService).passwordEncoder(passwordEncoder);
+        return authenticationManagerBuilder.getOrBuild();
+    }
     
     @Bean
     public FilterRegistrationBean<BasicAuthenticationFilter> basicAuthenticationFilter(

